sql/src/main/java/io/crate/planner/fetch/FetchPushDown.java

```diff
@@ -42,17 +42,56 @@
 
     private static class Context {
 
-        private final LinkedHashMap<Symbol, InputColumn> requiredInSub;
+        private final OrderBy orderBy;
+        private LinkedHashSet<Symbol> querySymbols;
+        private LinkedHashMap<Symbol, InputColumn> inputColumns;
 
-        private Context(LinkedHashMap<Symbol, InputColumn> sub) {
-            requiredInSub = sub;
+        private Context(@Nullable OrderBy orderBy) {
+            this.orderBy = orderBy;
+            if (orderBy != null) {
+                querySymbols = new LinkedHashSet<>(orderBy.orderBySymbols().size() + 1);
+                querySymbols.addAll(orderBy.orderBySymbols());
+            }
+        }
+
+        boolean isQuerySymbol(Symbol symbol) {
+            return querySymbols != null && querySymbols.contains(symbol);
+        }
+
+        boolean allocateQuerySymbol(Symbol symbol) {
```

Comment:
return value is never used ? name also doesn't indicate what the boolean would represent. Maybe just change it to void